lib/ansible/modules/cloud/podman/podman_container_info.py

```diff
@@ -0,0 +1,410 @@
+                        },
+                        {
+                            "Name": "RLIMIT_NPROC",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        }
+                    ],
+                    "CpuCount": 0,
+                    "CpuPercent": 0,
+                    "IOMaximumIOps": 0,
+                    "IOMaximumBandwidth": 0
+                }
+                }
+        ]
+"""
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_containers_facts(module, executable, name):
+    """Collect containers facts for all containers or for specified in 'name'.
+
+    Arguments:
+        module {AnsibleModule} -- instance of AnsibleModule
+        executable {string} -- binary to execute when inspecting containers
+        name {list} -- list of names or None in case of all containers
+
+    Returns:
+        list of containers info, stdout, stderr
+    """
+    if not name:
+        all_names = [executable, 'container', 'ls', '-q', '-a']
+        rc, out, err = module.run_command(all_names)
+        if rc != 0:
+            module.fail_json(msg="Unable to get list of containers: %s" % err)
+        name = out.split()
+        if not name:
+            return [], out, err
+    command = [executable, 'container', 'inspect']
+    command.extend(name)
+    rc, out, err = module.run_command(command)
+    if rc == 0:
+        json_out = json.loads(out) if out else None
+        if json_out is None:
+            return [], out, err
+        return json_out, out, err
+    if rc != 0 and 'no such ' in err:
+        if len(name) < 2:
+            return [], out, err
+        return cycle_over(module, executable, name)
+    module.fail_json(msg="Unable to gather info for %s: %s" % (",".join(name), err))
+
+
+def cycle_over(module, executable, name):
+    """Inspect each container in a cycle in case some of them don't exist.
+
+    Arguments:
+        module {AnsibleModule} -- instance of AnsibleModule
+        executable {string} -- binary to execute when inspecting containers
+        name {list} -- list of containers names to inspect
+
+    Returns:
+        list of containers info, stdout as empty, stderr
+    """
+    inspection = []
+    stderrs = []
+    for container in name:
+        command = [executable, 'container', 'inspect', container]
+        rc, out, err = module.run_command(command)
+        if rc != 0 and 'no such ' not in err:
+            module.fail_json(msg="Unable to gather info for %s: %s" % (container, err))
+        if rc == 0 and out:
+            json_out = json.loads(out)
+            if json_out:
+                inspection += json_out
+        stderrs.append(err)
+    return inspection, "", "\n".join(stderrs)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            executable=dict(type='str', default='podman'),
+            name=dict(type='list', elements='str')
```

Comment:
Use literals rather than constructors:
```suggestion
        argument_spec={
                'executable': {'type': 'str', 'default': 'podman'},
                'name': {'type': 'list', 'elements': 'str'},
        },
```

Reply:
What's the point of `'elements': 'str'` in this parameter? This is only used for sub options. This appears to be regular list parameter.

Reply:
> What's the point of `'elements': 'str'` in this parameter? This is only used for sub options. This appears to be regular list parameter.

`elements` can be used for every `type: 'list'` parameter. In this case, Ansible will make sure that the list contains strings (and not booleans, numbers, lists, dictionaries, ...).

Reply:
> Use literals rather than constructors:

Only 19 current modules use literals, other 667 use constructors:
```bash
$ grep -r 'argument_spec={' lib/ansible/modules/ | wc -l
19
$ grep -r 'argument_spec=dict' lib/ansible/modules/ | wc -l
667
```
Do we really want to break inconsistency? If it's a new rule, is it defined somewhere? Do we have a defined style at all that I can look at it? These style issues can be very easy caught by linter rules in `ansible-test` sanity check for example.
Just every time I do something like in most modules, it's always a way to do it different. I think all style rules should be defined in CI job to save people work to review functionality only. It also helps to keep the same style across the repo and different reviewers.

Reply:
This isn't really a new rule so much as a new suggestion. Constructors were the preference in the early days of Ansible and hence they have become prevalent. Literals are less confusing to those who are not familiar with constructors and they are usually faster (thought that's not as much an issue in module code).

For new modules, I suggest using literals rather than sticking with the (less than ideal) convention found in most modules.

Reply:
@felixfontein 
> elements can be used for every type: 'list' parameter.

I didn't know that. I've only ever seen this used for suboptions. I didn't know we handled it anywhere else. Now I know. 😄

Reply:
That used to be the case, but it changed somewhen this year (IIRC). I searched a bit, here's the PR which implemented validation: #50335